lib/libzfs/libzfs_pool.c

```diff
@@ -2925,6 +2926,9 @@ zpool_find_vdev(zpool_handle_t *zhp, const char *path, boolean_t *avail_spare,
 		verify(nvlist_add_uint64(search, ZPOOL_CONFIG_GUID, guid) == 0);
 	} else if (zpool_vdev_is_interior(path)) {
 		verify(nvlist_add_string(search, ZPOOL_CONFIG_TYPE, path) == 0);
+	} else if (path[0] != '/') {
+		(void) snprintf(buf, sizeof (buf), "/dev/%s", path);
+		verify(nvlist_add_string(search, ZPOOL_CONFIG_PATH, buf) == 0);
```

Comment:
Looking on the code in following chunk I have feeling that zfs_resolve_shortname() could be used here instead.  Am I wrong?

Reply:
The access checks in zfs_resolve_shortname() would cause this to fail at finding vdevs that are present in the config but missing from the system. I'm not sure if that's acceptable, but I can try it out.

Reply:
Passes the test suite, so I've incorporated this change.